### PR TITLE
Sonnet V1.2.3

### PR DESCRIPTION
**Pull request by ultrabear**
Not all the features are done but tbh its been so long since an update and theres some bugs that are only patched on dev, we can push stuff to V1.2.4

### CODE REVIEW

File: cmds/cmd_moderation.py [hunk 1]
```diff
@@ -498,6 +516,50 @@ async def grab_guild_message(message, args, client, **kwargs):
...
+            await message.channel.send("ERROR: Limit is not valid int")
+            return 1
+    else:
+        await message.channel.send("ERROR: No limit specified")
+        return 1
+
+    if limit > 100:
+        await message.channel.send("ERROR: Cannot purge more than 100 messages")
+        return 1
+
```
Comment (Saragon4005) [inline cmds/cmd_moderation.py:542]: what if the users put in 0 or negative numbers?

File: cmds/cmd_moderation.py [hunk 2]
```diff
@@ -599,7 +661,16 @@ async def grab_guild_message(message, args, client, **kwargs):
         'permission': 'moderator',
         'cache': 'keep',
         'execute': grab_guild_message
-        }
+        },
+    'purge':
+        {
+            'pretty_name': 'purge <limit> [user]',
+            'description': 'Purge messages from a given channel and optionally only from a specified user',
+            'rich_description': 'Can only purge up to 100 messages at a time to prevent catastrophic errors',
```
Comment (Saragon4005) [inline cmds/cmd_moderation.py:669]: this or point out this is a discord limitation, also maybe add that this can only delete 14 day or younger messages, although that another check and that's gonna be some kind of discord error, I think HTTPException, although the lib may handle it.

File: dlibs/dlib_messages.py [hunk 3]
```diff
@@ -256,9 +253,9 @@ async def on_message(message, **kargs):
     # Check message against automod
     stats["start-automod"] = round(time.time() * 100000)
 
-    spammer = antispam_check([message.channel.guild.id, message.author.id, message.created_at, ramfs, mconf["antispam"][0], mconf["antispam"][1]])
+    spammer = antispam_check(message.channel.guild.id, message.author.id, message.created_at, len(message.content), ramfs, int(float(mconf["antispam"][0])), float(mconf["antispam"][1]))
 
-    message_deleted = False
+    message_deleted: bool = False
```
Comment (Saragon4005) [inline dlibs/dlib_messages.py:258]: isn't this excessive?

File: dlibs/dlib_reactionroles.py [hunk 4]
```diff
@@ -46,17 +46,16 @@ async def on_raw_reaction_add(payload, **kargs):
...
-async def on_raw_reaction_remove(payload, **kargs):
+async def on_raw_reaction_remove(payload: discord.RawReactionActionEvent, **kargs: Any) -> None:
 
     if not payload.guild_id: return
 
     inc_statistics([payload.guild_id, "on-raw-reaction-remove", kargs["kernel_ramfs"]])
 
     client = kargs["client"]
-    rrconf = load_message_config(payload.guild_id, kargs["ramfs"], datatypes=reactionrole_types)["reaction-role-data"]
+    rrconf: Any = load_message_config(payload.guild_id, kargs["ramfs"], datatypes=reactionrole_types)["reaction-role-data"]
```
Comment (Saragon4005) [inline dlibs/dlib_reactionroles.py:56]: line 36 this was Dict what is going on here?

Comment (ultrabear) [reply dlibs/dlib_reactionroles.py:56]: line 36 used to be Any this was before the JSON cache type was made (that returns a pure pyobj)

File: dlibs/dlib_messages.py (hunk 3 above)
Comment (ultrabear) [reply dlibs/dlib_messages.py:258]: no :trollface:

File: cmds/cmd_moderation.py (hunk 2 above)
Comment (ultrabear) [reply cmds/cmd_moderation.py:669]: eh if people dont know how purge works thats their fault not mine

File: cmds/cmd_moderation.py (hunk 1 above)
Comment (ultrabear) [reply cmds/cmd_moderation.py:542]: oh dear

File: cmds/cmd_moderation.py [hunk 5]
```diff
@@ -283,31 +296,32 @@ async def mute_user(message, args, client, **kwargs):
...
 
     if kwargs["verbose"] and not mutetime:
-        await message.channel.send(f"Muted {member.mention} with ID {member.id} for {reason}")
+        await message.channel.send(f"Muted {member.mention} with ID {member.id} for {reason}", allowed_mentions=discord.AllowedMentions.none())
 
     if mutetime:
+
+        if not infractionID:
+            await message.channel.send("CAUGHT ERROR: There has been an error in grabbing the infractionID\n(User muted but no mute timer created)")
+            raise RuntimeError("Impossible code loop detected")
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply cmds/cmd_moderation.py:305]: this is impossible still but mypy got sad when i didint check if infraction id wasint a nonetype
tldr mypy needs branching type returns based on input args

File: Makefile [hunk 6]
```diff
@@ -1,6 +1,6 @@
 CC=gcc
-version=1.2.1-PATCH.0
+version=1.2.3-DEV.0
```
[the remark this answers is not in the document]
Comment (ultrabear) [reply Makefile:2]: it is moved versions, the Clib always stays at whatever version it was finalized at

[the remark this answers is not in the document]
Comment (Saragon4005) [reply Makefile:2]: so a -DEV version????

**Review by Saragon4005 — APPROVED** [review]

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (Saragon4005) [reply Makefile:2]: apperently
